code/game/objects/effects/alien_acid.dm

```diff
@@ -15,73 +15,89 @@
 	density = 0
 	opacity = 0
 	anchored = 1
-	unacidable = 1
-	var/atom/target
-	var/ticks = 0
-	var/target_strength = 0
+	acid_state = UNACIDABLE
+	var/acid_type = "alienacid"
+	var/turf/target
+	var/target_strength = 60 //phil235 roughly 1 minute for alien acid on wall, if 1 process per second
 
 
-/obj/effect/acid/New(loc, targ)
+/obj/effect/acid/New(loc, acid_pwr, acid_amt, acid_id)
 	..(loc)
-	target = targ
-
-	//handle APCs and newscasters and stuff nicely
-	pixel_x = target.pixel_x
-	pixel_y = target.pixel_y
-
-	if(isturf(target))	//Turfs take twice as long to take down.
-		target_strength = 640
-	else
-		target_strength = 320
-	tick()
 
+	target = get_turf(src)
 
-/obj/effect/acid/proc/tick()
-	if(!target)
-		qdel(src)
+	if(acid_id)
+		acid_type = acid_id
 
-	ticks++
+	if(acid_amt)
+		acid_level = min(acid_amt*acid_pwr, 12000)
 
-	if(ticks >= target_strength)
-		target.visible_message("<span class='warning'>[target] collapses under its own weight into a puddle of goop and undigested debris!</span>")
+	//handle APCs and newscasters and stuff nicely
+	pixel_x = target.pixel_x + rand(-4,4)
+	pixel_y = target.pixel_y + rand(-4,4)
 
-		if(istype(target, /obj/structure/closet))
-			var/obj/structure/closet/T = target
-			T.dump_contents()
-			qdel(target)
+	START_PROCESSING(SSobj, src)
 
-		if(istype(target, /turf/closed/mineral))
-			var/turf/closed/mineral/M = target
-			M.ChangeTurf(M.baseturf)
 
-		if(istype(target, /turf/open/floor))
-			var/turf/open/floor/F = target
-			F.ChangeTurf(F.baseturf)
+/obj/effect/acid/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	target = null
+	return ..()
 
-		if(istype(target, /turf/closed/wall))
-			var/turf/closed/wall/W = target
-			W.dismantle_wall(1)
+/obj/effect/acid/process()
+	if(!target)
+		qdel(src)
 
+	if(acid_type == "alienacid") //only alien acid can melt turfs
+		if(prob(50))
+			playsound(loc, 'sound/items/Welder.ogg', 100, 1)
```

Comment:
Since new acid can now damage you if you touch objects with acid on them or walk on turf with acid, I think it should be made clear the acid is "active". My fear is people running around aciding stuff left and right and people don't notice until it's too late and the stuff has melted.